core/domain/collection_services.py

```diff
@@ -928,39 +928,28 @@ def update_collection(
     collection = apply_change_list(collection_id, change_list)
 
     _save_collection(committer_id, collection, commit_message, change_list)
-    update_collection_summary(collection.id, committer_id)
+    regenerate_collection_summary(collection.id, committer_id)
 
     if (not rights_manager.is_collection_private(collection.id) and
             committer_id != feconf.MIGRATION_BOT_USER_ID):
         user_services.update_first_contribution_msec_if_not_set(
             committer_id, utils.get_current_time_in_millisecs())
 
 
-def create_collection_summary(collection_id, contributor_id_to_add):
+def regenerate_collection_summary(collection_id, contributor_id_to_add):
     """Creates and stores a summary of the given collection.
 
     Args:
         collection_id: str. ID of the collection.
-        contributor_id_to_add: str. ID of the contributor to be added to the
-            collection summary.
+        contributor_id_to_add: str|None. ID of the contributor to be added to
```

Comment:
In which case this will be None?

Reply:
If we want to regenerate the whole summary, which needs to be done in some cases.

Reply:
I have two questions:

1. We should not pass None as a `contributor_id`, it doesn't seems reasonable to have a user id as a None!
2. Why do the regenerate method takes this parameter, can we generate the sumarry without this id?

[Feel free to work on this comment in a separate PR!]

Reply:
#10758

Reply:
As per offline discussion, @vojtechjelinek is going to resolve this comment in a separate PR! We have an issue at #10758 to track these comments.